packages/workbox-routing/lib/Router.mjs

```diff
@@ -0,0 +1,328 @@
+/*
+ Copyright 2017 Google Inc. All Rights Reserved.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import core from 'workbox-core';
+import {_private} from 'workbox-core';
+
+import normalizeHandler from './normalizeHandler.mjs';
+
+/**
+ * The Router takes one or more [Routes]{@link Route} and allows you to apply
+ * that routing logic to determine the appropriate way of handling requests
+ * inside of a service worker.
+ *
+ * It also allows you to define a "default" handler that applies to any requests
+ * that don't explicitly match a `Route`, and a "catch" handler that responds
+ * to any requests that throw an exception while being routed.
+ *
+ * You can use the `handleRequest()` method to pass a `FetchEvent` through the
+ * router and ultimately get a "routed response" back.
+ * If you'd like this to be handled automatically for you, calling
+ * `addFetchListener()` will cause the `Router` to respond to `fetch` events.
```

Comment:
Two cents: I'd drop this in favor of explaining in the respective method descriptions.